controls/SV-230229.rb

```diff
@@ -23,64 +23,74 @@
-Root CA 3
-          Validity
-             Not Before: Mar 20 18:46:41 2012 GMT
-             Not After   : Dec 30 18:46:41 2029 GMT
-          Subject: C = US, O = U.S. Government, OU = DoD, OU = PKI, CN = DoD
-Root CA 3
-          Subject Public Key Info:
-             Public Key Algorithm: rsaEncryption
+Certificate:
+   Data:
+      Version: 3 (0x2)
+      Serial Number: 1 (0x1)
+      Signature Algorithm: sha256WithRSAEncryption
+      Issuer: C = US, O = U.S. Government, OU = DoD, OU = PKI, CN = DoD Root CA 3
+      Validity
+         Not Before: Mar 20 18:46:41 2012 GMT
+         Not After   : Dec 30 18:46:41 2029 GMT
+      Subject: C = US, O = U.S. Government, OU = DoD, OU = PKI, CN = DoD Root CA 3
+      Subject Public Key Info:
+         Public Key Algorithm: rsaEncryption
 
-    If the root ca file is not a DoD-issued certificate with a valid date and
-installed in the /etc/sssd/pki/sssd_auth_ca_db.pem location, this is a finding.
-  "
-  desc 'fix', "
-    Configure RHEL 8, for PKI-based authentication, to validate certificates by
-constructing a certification path (which includes status information) to an
-accepted trust anchor.
+If the root ca file is not a DoD-issued certificate with a valid date and installed in the /etc/sssd/pki/sssd_auth_ca_db.pem location, this is a finding.'
+  desc 'fix', 'Configure RHEL 8, for PKI-based authentication, to validate certificates by constructing a certification path (which includes status information) to an accepted trust anchor.
 
-    Obtain a valid copy of the DoD root CA file from the PKI CA certificate
-bundle from cyber.mil and copy the DoD_PKE_CA_chain.pem into the following file:
+Obtain a valid copy of the DoD root CA file from the PKI CA certificate bundle at cyber.mil and copy into the following file:
 
-    /etc/sssd/pki/sssd_auth_ca_db.pem
-  "
+/etc/sssd/pki/sssd_auth_ca_db.pem'
   impact 0.5
   tag severity: 'medium'
   tag gtitle: 'SRG-OS-000066-GPOS-00034'
-  tag satisfies: %w(SRG-OS-000066-GPOS-00034 SRG-OS-000384-GPOS-00167)
+  tag satisfies: ['SRG-OS-000066-GPOS-00034', 'SRG-OS-000384-GPOS-00167']
   tag gid: 'V-230229'
-  tag rid: 'SV-230229r627750_rule'
+  tag rid: 'SV-230229r858739_rule'
   tag stig_id: 'RHEL-08-010090'
-  tag fix_id: 'F-32873r567434_fix'
+  tag fix_id: 'F-32873r809269_fix'
   tag cci: ['CCI-000185']
-  tag nist: ['IA-5 (2) (a)']
+  tag nist: ['IA-5 (2) (a)', 'IA-5 (2) (b) (1)']
+  tag 'host', 'container'
+
+  only_if('If the System Administrator demonstrates the use of an approved alternate multifactor authentication method, this requirement is not applicable.', impact: 0.0) {
+    !input('smart_card_enabled')
+  }
 
-  ca_file = input('root_ca_file')
+  root_ca_loc = input('root_ca_location')
+  root_ca_file = input('root_ca_file')
+  root_ca_full_path = File.join(root_ca_loc, root_ca_file)
 
+  describe directory(root_ca_loc) do
+    it { should exist }
+  end
+
+  describe file(root_ca_full_path) do
+    it { should exist }
+  end
```

Comment:
These kinds of tests are not really necessary and if we need to check if a file exists to protect aginst a profile error it should be done in the code logic not as a test.